packages/plugin-ext/src/hosted/node/hosted-plugin-localization-service.ts

```diff
@@ -34,22 +36,38 @@ export interface VSCodeNlsConfig {
 }
 
 @injectable()
-export class HostedPluginLocalizationService {
+export class HostedPluginLocalizationService implements BackendApplicationContribution {
 
     @inject(LocalizationProvider)
     protected readonly localizationProvider: LocalizationProvider;
 
     @inject(EnvVariablesServer)
     protected readonly envVariables: EnvVariablesServer;
 
+    protected localizationDisposeMap = new Map<string, Disposable>();
     protected translationConfigFiles: Map<string, string> = new Map();
 
+    async initialize(): Promise<void> {
+        const cacheDir = await this.getLocalizationCacheDir();
+        await fs.emptyDir(cacheDir);
```

Comment:
Is it desirable to guarantee that this code completes before we start populating the cache directory in `buildLocalizations` / `deployLocalizations`? It seems like it always _should_, but it might still be a good idea to be sure.

Reply:
Actually, I just noticed that the backend server never awaits the `initialize` method, so we have to defer the execution of `buildTranslationConfig` until is has actually finished `emptyDir`. I added a small change which adds a `ready` promise to the service.